[DOC-2189] Add delegate selector details (#1554)
harness/developer-hub: docs/platform/2_Delegates/manage-delegates/select-delegates-with-selectors.md

helpdocs_is_published: true
---

> Harness runs tasks by using Harness Delegate to connect your environment to resources. Harness selects the best delegate based on previous use or round-robin selection. For more information, go to [How Does Harness Manager Pick Delegates?](/docs/platform/2_Delegates/delegate-concepts/delegate-overview.md#how-does-harness-manager-pick-delegates)

Reviewer: ```suggestion
Harness runs tasks by using a Harness delegate to connect your environment to your resources. Harness selects the best delegate based on previous connections or using a round-robin selection method. For more information, go to [How Does Harness Manager Pick Delegates?](/docs/platform/2_Delegates/delegate-concepts/delegate-overview.md#how-does-harness-manager-pick-delegates)
```

Reviewer 2: Left instance of Harness Delegate capitalized according to guidance here:
https://harness.atlassian.net/wiki/spaces/DOCS/pages/21130772574/Documentation+style+guide#Platform

Implemented suggested change.

In some cases, you might want Harness to select specific delegates. In these cases, you can use the **Delegate Selector** settings in Pipelines, Connectors, and so on, with corresponding delegate tags.

:::info note

> If no Delegates are selected for a CD step in its Delegate Selector setting, Harness prioritizes the Delegate used successfully for the Infrastructure Connector. For more information, go to [Which delegate is used during pipeline execution?](/docs/platform/delegates/delegate-concepts/delegate-overview/#which-delegate-is-used-during-pipeline-execution).

Reviewer: ```suggestion
If no delegates are selected for a CD step's **Delegate Selector** setting, Harness prioritizes the delegate used successfully for the infrastructure definition connector. For more information, go to [Which delegate is used during pipeline execution?](/docs/platform/delegates/delegate-concepts/delegate-overview/#which-delegate-is-used-during-pipeline-execution).
```
